src/controller_examples/rabbitmq_controller/spec/resource/common.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use crate::external_api::spec::*;
+use crate::kubernetes_api_objects::{
+    container::*, label_selector::*, pod_template_spec::*, prelude::*, resource_requirements::*,
+    volume::*,
+};
+use crate::kubernetes_cluster::spec::message::*;
+use crate::pervasive_ext::string_view::*;
+use crate::rabbitmq_controller::common::*;
+use crate::rabbitmq_controller::spec::types::*;
+use crate::reconciler::spec::{io::*, reconciler::*};
+use crate::state_machine::{action::*, state_machine::*};
+use crate::temporal_logic::defs::*;
+use vstd::prelude::*;
+use vstd::string::*;
+
+verus! {
+
+pub trait ResourceBuilder<T> {
```

Comment:
Could you add some documentation to explain the interfaces here? And how do you expect users to use it (I assume this is designed for workflow like: Get -> Create/Update)?

Reply:
This is because rabbitmq only has such flow; if more patterns are included, the interface can also be extended.

Reply:
And there are some things (about the design) to ponder